linkedList.js

```diff
@@ -3,6 +3,94 @@
  * @description  Main Module
  * @return {Object} an object exposing methods to be used to manipulate a linked list
  */
-function linkedListGenerator(){
 
-}
\ No newline at end of file
+function linkedListGenerator() {
+  let head = null;
+  let tail = null;
+  let nodeLength = 0;
+  return {
+    getHead: function() {
+      return head;
+    },
+    getTail: function() {
+      return tail;
+    },
+    add: function(val) {
+      let newNode = {};
+      newNode.value = val;
+      newNode.next = null;
+      if (!head) {
+        nodeLength = 1;
+        head = newNode;
+      }
+      if (tail) {
+        nodeLength++;
+        tail.next = newNode;
+        tail = newNode;
+      } else {
```

Comment:
This else wouldn't be necessary if you just update what tail is pointed to in the first if above. Also, the way that it stands, whenever you add to an empty list, your node length is actually 2 even though there's only one item in the list.